### PR TITLE
Teams

### PR DESCRIPTION
**Pull request by sbalay**
## Summary
- Create teams in organization
- Add teams to repositories
- Add members to teams

## Screenshots
screen shot 2017-08-29 at 7 57 10 pm

### CODE REVIEW

File: app/controllers/repositories.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const github = require('../services/github'),
 
 exports.new = (req, res, next) => {
   res.status(200);
-  res.render('index');
+  res.render('repositories');
```
Comment (mishuagopian) [inline app/controllers/repositories.js]: Good catch, lol

File: app/controllers/teams.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      res.status(200);
+      res.send({ team });
+    })
+    .catch(err => {
+      res.status(400);
+      res.send(err.internalError ? err : JSON.parse(err.message));
+    });
+};
+
+exports.addTeamToRepo = (req, res, next) => {
```
Comment (mishuagopian) [inline app/controllers/teams.js]: Hmmm, I'm pretty sure something is wrong here..

Comment (mishuagopian) [reply app/controllers/teams.js]: No team, no repo. What.

File: app/public/javascripts/teams.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+  }).then(function(res) {
+    teams = res.teams;
+    $('.teams-title').text('Teams');
+    addTeamButton.attr('disabled', false);
+    createTeamOptions(res.teams);
+  });
+
+  //
+  // $('.repository-create-button').click(function() {
```
Comment (mishuagopian) [inline app/public/javascripts/teams.js]: 🤔

Comment (mishuagopian) [reply app/public/javascripts/teams.js]: I used to have a friend who hated commented code. You may know him.

**Comment by sbalay** [comment]
@mishuagopian ready!

File: app/public/javascripts/vendor/jquery.growl.js [hunk 4]
```diff
@@ -0,0 +1,272 @@
+/* eslint-disable */
+
+/*
+jQuery Growl
```
Comment (mishuagopian) [inline app/public/javascripts/vendor/jquery.growl.js]: Why you adding this here and not just setting it as public from `app.js` file?

Comment (mishuagopian) [reply app/public/javascripts/vendor/jquery.growl.js]: Something like (using the corresponding file):
```node
  app.use('/jquery', express.static(`${__dirname}/node_modules/jquery/dist/`));
```

File: app/public/stylesheets/vendor/jquery.growl.css [hunk 5]
```diff
@@ -0,0 +1,85 @@
+/* jQuery Growl
```
Comment (mishuagopian) [inline app/public/stylesheets/vendor/jquery.growl.css]: Same here. You'll be setting the complete `jquery.growl` folder as public so requiring it from the html should be enough.

File: app/public/javascripts/vendor/jquery.growl.js (hunk 4 above)
Comment (sbalay) [reply app/public/javascripts/vendor/jquery.growl.js]: 😮

File: app/public/stylesheets/vendor/jquery.growl.css (hunk 5 above)
Comment (sbalay) [reply app/public/stylesheets/vendor/jquery.growl.css]: 😮 😮 😮

File: app/services/github.js [hunk 6]
```diff
@@ -146,6 +154,44 @@ exports.defaultTeams = settings => {
...
+  init(token);
+  return github.orgs
+    .getTeams({
+      org: config.common.github.organization,
+      per_page: 100,
+      page
+    })
+    .then(response => {
+      if (shouldFetchAnotherPage(response)) {
+        return exports.getPrivateRepos(token, page + 1).then(arr => arr.concat(response.data));
```
Comment (sbalay) [inline app/services/github.js]: 😱 😱 😱 😱 😱

File: app/controllers/teams.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    })
+    .catch(err => {
+      res.status(400);
+      res.send(err.internalError ? err : JSON.parse(err.message));
+    });
+};
+
+exports.addMemberToTeam = (req, res, next) => {
+  const { teamId, username } = req.params;
+  const { token, maintainer } = req.body;
```
Comment (sbalay) [inline app/controllers/teams.js]: known issue, this body params is a string but is used like if it was boolean

File: app/public/javascripts/vendor/jquery.growl.js (hunk 4 above)
Comment (mishuagopian) [reply app/public/javascripts/vendor/jquery.growl.js]: What about this?

File: app/public/stylesheets/vendor/jquery.growl.css (hunk 5 above)
Comment (mishuagopian) [reply app/public/stylesheets/vendor/jquery.growl.css]: And this

**Pull request merged**